### DISCUSSION TITLE
Question: is it possible to create (using the action button maybe) a 'select 2 out of X' input switch for the image compare from RGthree?

### DISCUSSION THREAD

**Discussion by stephantual** [rgthree/rgthree-comfy#148] [Q&A]
A picture paints a thousand words: 
![image](https://github.com/rgthree/rgthree-comfy/assets/1596070/0f7f56d8-d539-4614-8ef6-5dddf909c796)

I'm doing this demo tutorial on how to use SDXL 'properly'. In order to demo the outcome, I generate 3+ outputs with various techniques. So far so good. Now i want to compare the results. So i use the excellent Rgthree 'image compare' but.. it only takes TWO inputs (evidently). 

Let's imagine I have three outcomes refered to as A, B and C. 
How would I organize my workflow using RGthree to be able to quickly select "A vs C" or "B vs C" or "A vs B" (you get the point).

The image i uploaded is simplified on purpose, the 'real' output has dozens of potential outputs, hence why i think Rghtree woudl be useful here, but I can't wrap my head around a 'clean' way to do this.

Any pointers, worflow examples etc would be appreciated. Thank you!

**Comment by OmanovicMiran** [comment]
I kind of cheesed this to achieve somewhat what you are asking for. I my case i have all 9 samplers groups (similar to your 3) to compare effect of sampler (everything else being same)
1. Image output from each VAE encoder I connected to set node.  In your case you can set  [A], [B], [C].
2. I named the groups SCH - [A], SCH [B] ...
3. I added group that catches all these individual groups and i named it SCH - ALL.
4.  Then off screen i used get for each of the individual groups.
5. I added 2 Any switch (rgthree) which return first enabled (not bypased) from tthe provided.
6. I connected first anyswitch ABC and second CBA. So it will show first enabled from front and first enabled from back.
7. Output of those i connected to image comparer.
8. I added Fast Groups Mutter and filtered the by "SCH".

This would show all and than each individual group. All allows you to toggle off all of them quickly and you have toggles for each individual one.

Sure there is no validation if none are selected. If one is toggled comparer will compare the image to itself. When 2 toggled it works perfectly. When 3 or more toggled it will take first and last ignore middle selection. 

Note: JPS-Nodes has image switch node but i dont like the UI where you select the number. With my setup you kinda can use name and number.

I think i might create a custom node for this that would allow you to select compare images sometimes.

**Reply by OmanovicMiran** [reply]
Once i tested it more it requeues the VAE Decode because of the mute change. Tried it with fast group bypasser instead.

**Comment by Symbiomatrix** [comment]
I dunno about rgthree, but here's how I would do it in the laziest way possible:
- Impact pack has a node called "make list (any)" or (impact)makelistany - it takes in any number of uniform type inputs (expanding as you add more), and converts them into a list of that type.
- Create a custom node which given a list of any type and two widget ints, returns the list values at the given indices. Since the number of outputs and widgets is fixed, doesn't need any fancy js (and dabbling with the js code for input / output control is a major undertaking) - it's all of 20 lines of code to retrieve two generic elements from a list.

It's also called a multiplexer, ~[this repo might have one.](https://github.com/glibsonoran/Plush-for-ComfyUI)~ Not great.
